Build fails with java 1.8

```
Build fails with java 1.8:

mod.rng-parse.compile-main:
[...]
    [javac] /builddir/build/BUILD/jing-trang-20131210/build/mod/rng-parse/gensrc/main/com/thaiopensource/relaxng/parse/compact/CompactSyntax.java:1590: error: incompatible types: Annotations cannot be converted to AnnotationsImpl
    [javac]       a = IncludeBody(include, scope, a);
    [javac]                      ^
    [javac]   where AnnotationsImpl,Location,ElementAnnotation,CommentListImpl are type-variables:
    [javac]     AnnotationsImpl extends Annotations<Location,ElementAnnotation,CommentListImpl> declared in class CompactSyntax
    [javac]     Location extends Object declared in class CompactSyntax
    [javac]     ElementAnnotation extends Object declared in class CompactSyntax
    [javac]     CommentListImpl extends CommentList<Location> declared in class CompactSyntax
    [javac] Note: /builddir/build/BUILD/jing-trang-20131210/build/mod/rng-parse/gensrc/main/com/thaiopensource/relaxng/parse/compact/CompactSyntax.java uses unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 1 error
    [javac] 3 warnings
BUILD FAILED

```

Original issue reported on code.google.com by `vsky...@gmail.com` on 9 Jun 2014 at 7:38

--- a/mod/rng-parse/src/main/com/thaiopensource/relaxng/parse/compact/CompactSyntax.jj
+++ b/mod/rng-parse/src/main/com/thaiopensource/relaxng/parse/compact/CompactSyntax.jj
@@ -990,7 +990,7 @@ void Include(GrammarSection<Pattern, Location, ElementAnnotation, CommentListImp
   Token t;
   String href;
   String ns;
-  Include include = section.makeInclude();
+  Include<Pattern, Location, ElementAnnotation, CommentListImpl, AnnotationsImpl> include = section.makeInclude();
 }
 {
   t = "include" href = Literal()

